core/src/Telemetry/XMsTelemetryInfo.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System.Globalization;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Identity.Core.Telemetry
+{
+    internal class XMsTelemetryInfo
+    {
+        public string Version { get; set; }
+        public string ServerErrorCode { get; set; }
+        public string ServerSubErrorCode { get; set; }
+        public string TokenAge { get; set; }
+        public string SpeInfo { get; set; }
+    }
+
+    internal static class XMsTelemInfoHelper
+    {
+        internal static XMsTelemetryInfo parseXMsTelemHeader(string headerValue, RequestContext requestContext)
+        {
+            if (string.IsNullOrEmpty(headerValue))
+            {
+                return null;
+            }
+
+            string[] headerSegments = headerValue.Split(',');
+            if (headerSegments.Length == 0)
+            {
+                requestContext.Logger.Warning("Malformed x-ms-clitelem header");
```

Comment:
You should pull these strings into constants as they will help you test./

Reply:
done